lib/app/utils/hl7_validator.rb

```diff
@@ -3,6 +3,11 @@
 module Inferno
   # A validator that calls out to the HL7 validator API
   class HL7Validator
+    DETAILS_FILTER = [
+      %r{Sub-extension url 'introspect' is not defined by the Extension http://fhir-registry.smarthealthit.org/StructureDefinition/oauth-uris},
+      %r{Sub-extension url 'revoke' is not defined by the Extension http://fhir-registry.smarthealthit.org/StructureDefinition/oauth-uris},
+      /URL value .* does not resolve/
```

Comment:
I pasted `http://fhir-registry.smarthealthit.org/StructureDefinition/oauth-uris` into my browser and got a 404, is this expected? I don't know too much about these filters, but is it worth combining the first two Regexs by replacing them with `%r{Sub-extension url '(introspect|revoke)' is not defined by the Extension http://fhir-registry.smarthealthit.org/StructureDefinition/oauth-uris}`? Are there other possible values that could go in those quotes?

Reply:
I'm guessing that the URL doesn't lead anywhere because it's a canonical URL, is that correct?

Reply:
Some other relevant changes to the regexes might be to escape the `.` characters with `\.` and to begin and end the regexes with `^` and `$`. I'm not saying these changes have to be made, I'm just making sure we're considering them

Reply:
`http://fhir-registry.smarthealthit.org/StructureDefinition/oauth-uris` not resolving is ok because the validator can already has the definition or can get it through another means like the package server.

It is related to this [chat.fhir conversation](https://chat.fhir.org/#narrow/stream/179170-smart/topic/OAuth-URIs.20Extension) where the definition is different in SMART than in base FHIR.  we've seen issues where the validator picks up the one from the base FHIR spec which isn't correct.

Reply:
We don't want to escape `.` here because `.*` is saying "zero or more of any single character"

Reply:
Well you want to escape some of them, but not `.*` (probably what Jason meant).

Reply:
Oops, yeah I didn't mean `.*`, but the ones in the url

Reply:
%r{} escapes all special character. /a\.b/ is the same as %r{a.b}

Reply:
The problem is that this will match more than just that string, so to be thorough you should escape `.`, and prevent it from having any characters before or after the string, etc.  I think it is good to be thorough, but if you know if matches at least the string you want I also think that there is low likelyhood you will match something you don't want to match in this case.

Reply:
I don't think that's correct @yunwwang, I just tested and `%r{}` does not escape periods (it looks like it only escapes `/`.  @jason-crowley is right those need to be escaped.

I went into the `hl7_validator_operation_outcome.json` and changed the `.` to a random character and the tests still passed even though they should have failed (i.e. it is considering the `.` as any single character)

Reply:
fixed